src/flake.rs

```diff
@@ -178,9 +186,15 @@ impl<'de> Deserialize<'de> for FlakeLock {
                     panic!("root node was not a Root node, but was a {} node", root_node.variant());
                 };
                 for (root_name, root_reference) in root_node.inputs.iter() {
-                    let root_reference = root_reference.as_str();
-                    let root_node = nodes[root_reference].to_owned();
-                    root_nodes.insert(root_name.to_owned(), root_node);
+                    let node_keys: Vec<String> = match root_reference {
+                        Input::String(s) => vec![s.to_string()],
+                        Input::List(keys) => keys.to_owned(),
+                    };
+
+                    for key in node_keys {
+                        let root_node = nodes[&key].to_owned();
+                        root_nodes.insert(root_name.to_owned(), root_node);
+                    }
```

Comment:
I don't think you've interpreted the meaning of the multiple keys correctly. It's a hierarchical list of keys, not a flat one. The flake format documentation in `nix flake --help` says,

> It is also possible to "inherit" an input from another input. This is useful to minimize flake
> dependencies. For example, the following sets the nixpkgs input of the top-level flake to be
> equal to the nixpkgs input of the dwarffs input of the top-level flake:
>
>   | inputs.nixpkgs.follows = "dwarffs/nixpkgs";
>
> The value of the follows attribute is a /-separated sequence of input names denoting the path of
> inputs to be followed from the root flake.

Reply:
The example we've been working with has this in the flake:

```nix
inputs.nixpkgs.follows = "haskellNix/nixpkgs-unstable";
```

which produces this in the lock:

```json
{
  "inputs": {
    "nixpkgs": [
      "haskellNix",
      "nixpkgs-unstable"
    ],
  }
}
```

Reply:
I'm currently in the process of researching this and I have this PR marked as a draft for precisely that reason.